client/components/async-load/index.jsx

```diff
@@ -32,12 +39,12 @@ export default class AsyncLoad extends Component {
 
 	require() {
 		this.props.require( ( component ) => {
-			this.setState( { component } );
+			this.setState( { component, dirty: false } );
```

Comment:
We might also want to check that the current `require` prop matches what it was when we started async load of the component:

```js
const { require: requireComponent } = this.props;
requireComponent( ( component ) => {
	if ( requireComponent === this.props.require ) {
		this.setState( { component } );
	}
} );
```